rawkv/rawkv.go

```diff
@@ -176,7 +198,7 @@ func (c *Client) Get(ctx context.Context, key []byte, options ...RawOption) ([]b
 	req := tikvrpc.NewRequest(
 		tikvrpc.CmdRawGet,
 		&kvrpcpb.RawGetRequest{
-			Key: key,
+			Key: c.buildRequestKey(key),
```

Comment:
Then we will need to encode keys twice (one for request PD & another for request TiKV) ? How about do the encoding before here. And then it also seems that we don't need the `CodecPDClientV2`.

Reply:
Yes, we will encode keys twice. I keep the encoding happening only in the request sending phase because I don't want to change the existing logic of the interface like `Scan`/`RawScan`. The current commit doesn't delay this encoding late enough, in the next one, I am trying to encode the key and set the API version in `kvrpcpb.Context` only in package `tikvrpc`. This might keep the encoding transparent from the upper code, reducing the appearance of bugs.

Reply:
Besides, the API version is hard to pass to some request constructors like in struct `Scanner`. Keeping this information only in `RegionRequestSender` might save us from the code review hell.